### DISCUSSION TITLE
Shoudl CMakeLists be updated for intel oneAPI?

### DISCUSSION THREAD

**Discussion by ghylander** [NatLabRockies/ROSCO#106] [General]
I saw in the openFAST repository that in pull request [727](https://github.com/OpenFAST/openfast/pull/727), the cmake options file for fortran compilation was updated for "for Intel OneAPI compiler detection"

My team and I have no trouble compiling ROSCO with oneAPI using the current CMakeLists file. Also i'm not 100% sure of how the changes account for oneAPI, seems like the changes where mostly replacing:
`if (CMAKE_Fortran_COMPILER_ID STREQUAL "Intel")`
with:
`if (${CMAKE_Fortran_COMPILER_ID} MATCHES "^Intel")`

**Comment by rafmudaf** [comment]
Hi @ghylander

CMake lists the ID's they use for different compiler vendors [here](https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_COMPILER_ID.html). There are two Intels: "Intel" and "IntelLLVM". The new Fortran compiler (`ifx`) will be IntelLLVM whereas the classic Intel Fortran compiler (`ifort`) will be simply Intel. Note that ifort is distributed with oneAPI, but it does not use the new LLVM backend. If the ROSCO CMake is working fine, you are likely using ifort and the existing CMake setup is good to go.

**Reply by ghylander** [reply]
Ah I see, thanks for the info, quite helpful
I guess it can stay as it is for now then

**Reply by rafmudaf** [reply]
> I guess it can stay as it is for now then

Well, this is a good point... it will need to be addressed at some point. I haven't actually been able to use the `ifx` compiler myself as it seems to be missing from the oneAPI packages I get. If someone is able to use it, it would be good to test with CMake configuration here as well as in OpenFAST.
